client/src/pages/ProfileDetails/ProfileDetails.tsx

```diff
@@ -1,62 +1,92 @@
+import { useState, useEffect, SyntheticEvent } from 'react';
 import { Grid, Paper, Typography } from '@material-ui/core/';
 import { Rating, Button } from '@mui/material';
 import { createTheme, ThemeProvider } from '@mui/material/styles';
-import { useState } from 'react';
 import useStyles from './useStyles';
+import { useParams } from 'react-router-dom';
 import LocationOnIcon from '@mui/icons-material/LocationOn';
 import TextField from '@mui/material/TextField';
 import AdapterDateFns from '@mui/lab/AdapterDateFns';
 import LocalizationProvider from '@mui/lab/LocalizationProvider';
 import DateTimePicker from '@mui/lab/DateTimePicker';
-import { mockProfile } from '../../mocks/mockProfile';
-export default function Profile(): JSX.Element {
+import { profileGet } from '../../helpers/APICalls/profile';
+import { Profile } from '../../interface/Profile';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { sendRequest } from '../../helpers/APICalls/request';
+interface ProfileParams {
+  id: string;
+}
+export default function ProfileDetails(): JSX.Element {
+  const { updateSnackBarMessage } = useSnackBar();
   const classes = useStyles();
-  const [dropInValue, setDropInValue] = useState<Date | null>(new Date());
-  const [dropOffValue, setDropOffValue] = useState<Date | null>(new Date());
-  const { firstName, lastName, introduction, address, description, galleryPics, profilePic, coverPic } = mockProfile;
+  const { id } = useParams<ProfileParams>();
+  const initialData: Profile = {
+    firstName: '',
+    lastName: '',
+    address: '',
+    description: '',
+    gender: 'male',
+  };
+  const [profile, setProfile] = useState<Profile | null | undefined>(initialData);
+  const [start, setDropInValue] = useState<string | null | undefined>('');
+  const [end, setDropOffValue] = useState<string | null | undefined>('');
+  const handleSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+    sendRequest({ id, start, end }).then((data) => {
```

Comment:
The request controller expects the Id of a user not a profile, you can modify that by simply substituting id with `profile.userId` - you already saved the profile with `usestate`

Your modification breaks this endpoint for any other feature that sends the `userId` e.g `manageBookings UI` 🙂